### DISCUSSION TITLE
Config composition without using hydra

### DISCUSSION THREAD

**Discussion by JamesLarkinWhite** [omry/omegaconf#1078] [Q&A]
Dear all,

is something like Config composition as  is solved by Hydra using 
also possible by just using Omegaconf ? I would like to avoid the additional 
overhead of hydra and find this feature very usefull.

https://github.com/omry/omegaconf/issues/541

Thank you in advance

**Comment by Jasha10** [comment]
Hi @JamesLarkinWhite,

Hydra is a framework that automates many steps in the config composition process. Under the hood, Hydra drives OmegaConf to load and merge configs.

You can do config composition using pure OmegaConf -- you'll have to handle the business logic of loading and merging the configs yourself. If you want to achieve something like Hydra's "defaults list": this is something you'd need to code up yourself.

**Comment by omry** [comment]
You can use `OmegaConf.merge()` or direct assignment yourself.
However, I do recommend that you use Hydra, or at least the Hydra Compose API instead of inventing your own system.
